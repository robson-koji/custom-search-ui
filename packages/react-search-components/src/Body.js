import React from "react";

export default function Body({
  children,
  sideContent,
  bodyContent,
  bodyHeader,
  bodyFooter
}) {
  return (
    <div className="reference-ui-body">
      <div className="search-results">
        <div className="sidebar">{sideContent}</div>
        <div className="results">
          <div className="results__header">
            <div className="meta">{bodyHeader}</div>
          </div>
          <div className="results__body">{children || bodyContent}</div>
          <div className="results__footer">{bodyFooter}</div>
        </div>
      </div>
    </div>
  );
}
