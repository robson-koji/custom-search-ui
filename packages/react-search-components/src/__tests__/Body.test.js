import React from "react";
import Body from "../Body";
import { shallow } from "enzyme";

it("renders correctly", () => {
  const wrapper = shallow(<Body />);
  expect(wrapper).toMatchSnapshot();
});
