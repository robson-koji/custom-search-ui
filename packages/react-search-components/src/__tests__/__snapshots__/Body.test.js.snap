// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Body />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="search-results"
      >
        <div
          className="sidebar"
        />
        <div
          className="results"
        >
          <div
            className="results__header"
          >
            <div
              className="meta"
            />
          </div>
          <div
            className="results__body"
          />
          <div
            className="results__footer"
          />
        </div>
      </div>,
      "className": "reference-ui-body",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="sidebar"
          />,
          <div
            className="results"
          >
            <div
              className="results__header"
            >
              <div
                className="meta"
              />
            </div>
            <div
              className="results__body"
            />
            <div
              className="results__footer"
            />
          </div>,
        ],
        "className": "search-results",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "sidebar",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="results__header"
              >
                <div
                  className="meta"
                />
              </div>,
              <div
                className="results__body"
              />,
              <div
                className="results__footer"
              />,
            ],
            "className": "results",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="meta"
                />,
                "className": "results__header",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "meta",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "results__body",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "results__footer",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="search-results"
        >
          <div
            className="sidebar"
          />
          <div
            className="results"
          >
            <div
              className="results__header"
            >
              <div
                className="meta"
              />
            </div>
            <div
              className="results__body"
            />
            <div
              className="results__footer"
            />
          </div>
        </div>,
        "className": "reference-ui-body",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="sidebar"
            />,
            <div
              className="results"
            >
              <div
                className="results__header"
              >
                <div
                  className="meta"
                />
              </div>
              <div
                className="results__body"
              />
              <div
                className="results__footer"
              />
            </div>,
          ],
          "className": "search-results",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "sidebar",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="results__header"
                >
                  <div
                    className="meta"
                  />
                </div>,
                <div
                  className="results__body"
                />,
                <div
                  className="results__footer"
                />,
              ],
              "className": "results",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <div
                    className="meta"
                  />,
                  "className": "results__header",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                    "className": "meta",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "div",
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "results__body",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "results__footer",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
